kiosk/sources/extensions/kiosk_metadata.move

```diff
@@ -0,0 +1,144 @@
+        // verifies that we have kiosk ownership and returns the kiosk's ID.
+        // exits if the user doesn't have write access.
+        let mut_kiosk_uid = getWriteableKioskId(kiosk, cap);
+
+        // remove the Metadata from the kiosk.
+        let _ : Metadata = df::remove(mut_kiosk_uid, Kiosk_MetadataExt{}); 
+    }
+
+    // === Data Modifiers ===
+    // 
+    // Sets a custom `name` field with the `value`.
+    entry public fun add(kiosk: &mut Kiosk, cap: &KioskOwnerCap, name: String, value: String
+        ) {
+
+        // gets writeable kiosk Id if it exists, otherwise exits with error.
+        let mut_kiosk_uid = getWriteableKioskId(kiosk, cap);
+        // gets mutable metadata object if it's registered, otherwise exits with error.
+        let metadata = get_mut_metadata(mut_kiosk_uid); 
+
+        add_internal(metadata, name, value)
+    }
+
+    // Sets multiple `fields` with `values`.
+    entry public fun add_multiple(kiosk: &mut Kiosk, cap: &KioskOwnerCap, fields: vector<String>, values: vector<String>
+    ) {
+        // gets writeable kiosk Id if it exists, otherwise exits with error.
+        let mut_kiosk_uid = getWriteableKioskId(kiosk, cap);
+
+        // gets mutable metadata object if it's registered, otherwise exits with error.
+        let metadata = get_mut_metadata(mut_kiosk_uid); 
+
+        let len = vector::length(&fields);
+        assert!(len == vector::length(&values), EVecLengthMismatch);
+
+        let i = 0;
+        while (i < len) {
+            add_internal(metadata, *vector::borrow(&fields, i), *vector::borrow(&values, i));
+            i = i + 1;
+        };
+    }
+
+     // Remove a key from the metadata vec map.
+    entry public fun remove_key(kiosk: &mut Kiosk, cap: &KioskOwnerCap, name: String) {
+
+        // gets writeable kiosk Id if it exists, otherwise exits with error.
+        let mut_kiosk_uid = getWriteableKioskId(kiosk, cap);
+
+        // gets mutable metadata object if it's registered, otherwise exits with error.
+        let metadata = get_mut_metadata(mut_kiosk_uid); 
+
+        // verify that the key we're trying to remove exists on the metadata.
+        assert!(vec_map::contains(&metadata.metadata, &name), EKeyNotExists);
+
+        // remove key from map
+        vec_map::remove(&mut metadata.metadata, &name);
+    }
+
+    // === Access data ==== 
+    // Get the metadata for the kiosk
+    public fun get_metadata(kiosk_uid: &UID): &Metadata {
+        df::borrow(kiosk_uid, Kiosk_MetadataExt{})
+    }
+
+    // Access to the metadata VecMap
+    public fun get_metadata_vecmap(metadata: &Metadata): &VecMap<String, String>{
+        &metadata.metadata
+    }
+
+    // === internal helpers ====
+
+    // Internal helper to verify read-access to the kiosk
+    // Returns the `kiosk_id` if the user has write access
+    // Exits if kioskOwnerCap is invalid (or doesn't match the kiosk object)
+    fun getWriteableKioskId(kiosk: &mut Kiosk, cap: &KioskOwnerCap): &mut UID{
```

Comment:
We don't use `camelCase` for function names in Move, current convention is to use `snake_case`.